CorsixTH/Lua/epidemic.lua

```diff
@@ -262,13 +262,16 @@ function Epidemic:checkNoInfectedPlayerHasLeft()
         px and py and not self.hospital:isInHospital(px,py) then
       -- Patient escaped from the hospital, discovery is inevitable.
       self.result_determined = true
-      if not self.inspector then
-        self:spawnInspector()
-      end
       self:finishCoverUp()
       return
     end
   end
+
+  local all_infected_cured = self:countInfectedPatients() == 0
+  if all_infected_cured then
+    self:finishCoverUp()
+    return
+  end
```

Comment:
Function this is inserted in does not fit its current purpose.

The function name either needs updating; or this check could go elsewhere. Judging how the the tick function is set up, you'd make it a function in itself. First option better this time round though.

The name `all_infected_cured` is also potentially not a true reflection of the check you're running. We don't know that every patient got cured, one might have died instead.

Reply:
> Function this is inserted in does not fit its current purpose.
> 
> The function name either needs updating; or this check could go elsewhere. Judging how the the tick function is set up, you'd make it a function in itself. First option better this time round though.
> 
> The name `all_infected_cured` is also potentially not a true reflection of the check you're running. We don't know that every patient got cured, one might have died instead.

Yeap, agree.
I have moved new check to separated function.

Also I agree that `all_infected_cured` not reflecting the thing it contains. I have renamed it to `no_any_infected_patients_left`.

Also from my point of view we should rename function `checkNoInfectedPlayerHasLeft` to something more understandable. For example `checkIsAnyInfectedPatientLeftTheHospitalAndRevealedEpidemic`.

What do you think about this?

Reply:
> `checkNoInfectedPlayerHasLeft`

Players also get infected?  :O

In general, the name of the function indicates what you can expect to find inside.

The function name here is quite specific. For a larger function this is fine, as you mostly need all space for implementing the functionality. (Eg if you need 300 lines of code for that check, it's unlikely to be a good spot for doing something else as well.)

For a short function (which seems to be the case here), it is often more useful to give a slightly more general name. In that way, you sort-of pave the path for adding more checks or doing more work in that function in the future.
For example, if you use "isEpidemicContained" or "isEpidemicUnderControl" for example, then basically the function name suggests it covers any action that must be done when the epidemic "has finished". It could for example call off all nurses that are doing vaccination patrol in the corridor, etc.

The suggested function name is too narrow in scope for the latter action ("hasAnyPatientLeft" does not intuitively suggest actions wrt nurses), so if nurses need to be called off somewhen in the future development, it must be done elsewhere or the name of the function must be adapted.